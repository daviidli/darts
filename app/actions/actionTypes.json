{
	"SET_SERIAL_PORT": "SET_SERIAL_PORT",
	"SET_WAIT_TIME": "SET_WAIT_TIME",
	"SET_PLAYERS": "SET_PLAYERS",
	"ADD_PLAYER": "ADD_PLAYER",
	"REMOVE_PLAYER": "REMOVE_PLAYER",
	"CHANGE_PLAYER": "CHANGE_PLAYER",
	"SET_MAX_DARTS": "SET_MAX_DARTS",
	"SET_MAX_ROUNDS": "SET_MAX_ROUNDS",
	"SET_CURRENT_PLAYER": "SET_CURRENT_PLAYER",
	"SET_CURRENT_DART": "SET_CURRENT_DART",
	"SET_CURRENT_ROUND": "SET_CURRENT_ROUND",
	"INITIALIZE_ROUNDS": "INITIALIZE_ROUNDS",
	"SET_ROUND": "SET_ROUND",
	"SET_TOTALS": "SET_TOTALS",
	"SET_WINNER": "SET_WINNER",
	"SET_WAITING": "SET_WAITING"
}
