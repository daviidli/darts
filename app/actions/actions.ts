import actions from './actions.json';

export const setSerialPort = (port: string) => ({
	type: actions.SET_SERIAL_PORT,
	port
});

export const test = () => {
	return {};
};
