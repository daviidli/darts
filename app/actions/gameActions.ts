import actions from './actionTypes.json';

export const setPlayers = (players: string[]) => ({
	type: actions.SET_PLAYERS,
	players
});

export const setPlayer = (player: string, index: number) => ({
	type: actions.SET_PLAYER,
	player,
	index
});

export const addPlayer = (player: string) => ({
	type: actions.ADD_PLAYER,
	player
});

export const removePlayer = (player: string) => ({
	type: actions.REMOVE_PLAYER,
	player
});

export const setCurrentPlayer = (index: number) => ({
	type: actions.SET_CURRENT_PLAYER,
	index
});

export const setMaxRounds = (maxRounds: number) => ({
	type: actions.SET_MAX_ROUNDS,
	maxRounds
});

// todo: set rounds

export const addRound = (round: string[]) => ({
	type: actions.ADD_ROUND,
	round
});

export const setMaxThrows = (maxThrows: number) => ({
	type: actions.SET_MAX_THROWS,
	maxThrows
});

export const setCurrentRound = (currentRound: number) => ({
	type: actions.SET_CURRENT_ROUND,
	currentRound
});

export const setCurrentThrow = (currentThrow: string[]) => ({
	type: actions.SET_CURRENT_THROW,
	currentThrow
});

export const setTotals = (totals: number[]) => ({
	type: actions.SET_TOTALS,
	totals
});
