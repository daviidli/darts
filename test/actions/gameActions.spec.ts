import * as gameActions from '../../app/actions/gameActions';

describe('actions', () => {
	it('should create setPlayers action', () => {
		const players = [
			{ id: '1', name: 'player1' },
			{ id: '2', name: 'player2' }
		];
		expect(gameActions.setPlayers(players)).toMatchSnapshot();
	});

	it('should create addPlayer action', () => {
		expect(gameActions.addPlayer('player1')).toMatchSnapshot();
	});

	it('should create removePlayer action', () => {
		expect(gameActions.removePlayer(0)).toMatchSnapshot();
	});

	it('should create changePlayer action', () => {
		expect(gameActions.changePlayer('unicorn', 0)).toMatchSnapshot();
	});

	it('should create setMaxDarts action', () => {
		expect(gameActions.setMaxDarts(3)).toMatchSnapshot();
	});

	it('should create setMaxRounds action', () => {
		expect(gameActions.setMaxRounds(12)).toMatchSnapshot();
	});

	it('should create setCurrentPlayer action', () => {
		expect(gameActions.setCurrentPlayer(1)).toMatchSnapshot();
	});

	it('should create setCurrentDart action', () => {
		expect(gameActions.setCurrentDart(1)).toMatchSnapshot();
	});

	it('should create setCurrentRound action', () => {
		expect(gameActions.setCurrentRound(0)).toMatchSnapshot();
	});

	it('should create initializedRound action', () => {
		expect(gameActions.initializedRound([[['']]])).toMatchSnapshot();
	});

	it('should create setRound action', () => {
		expect(gameActions.setRound(0, 0, 0, '20-t')).toMatchSnapshot();
	});

	it('should create setTotals action', () => {
		expect(gameActions.setTotals([301])).toMatchSnapshot();
	});

	it('should create setWinner action', () => {
		expect(gameActions.setWinner(0)).toMatchSnapshot();
	});

	it('should create setWaiting action', () => {
		expect(gameActions.setWaiting(true)).toMatchSnapshot();
	});
});
