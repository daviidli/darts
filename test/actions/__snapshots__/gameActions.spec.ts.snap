// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create addPlayer action 1`] = `
Object {
  "player": "player1",
  "type": "ADD_PLAYER",
}
`;

exports[`actions should create changePlayer action 1`] = `
Object {
  "index": 0,
  "player": "unicorn",
  "type": "CHANGE_PLAYER",
}
`;

exports[`actions should create initializedRound action 1`] = `
Object {
  "rounds": Array [
    Array [
      Array [
        "",
      ],
    ],
  ],
  "type": "INITIALIZE_ROUNDS",
}
`;

exports[`actions should create removePlayer action 1`] = `
Object {
  "index": 0,
  "type": "REMOVE_PLAYER",
}
`;

exports[`actions should create setCurrentDart action 1`] = `
Object {
  "index": 1,
  "type": "SET_CURRENT_DART",
}
`;

exports[`actions should create setCurrentPlayer action 1`] = `
Object {
  "index": 1,
  "type": "SET_CURRENT_PLAYER",
}
`;

exports[`actions should create setCurrentRound action 1`] = `
Object {
  "index": 0,
  "type": "SET_CURRENT_ROUND",
}
`;

exports[`actions should create setMaxDarts action 1`] = `
Object {
  "max": 3,
  "type": "SET_MAX_DARTS",
}
`;

exports[`actions should create setMaxRounds action 1`] = `
Object {
  "max": 12,
  "type": "SET_MAX_ROUNDS",
}
`;

exports[`actions should create setPlayers action 1`] = `
Object {
  "players": Array [
    Object {
      "id": "1",
      "name": "player1",
    },
    Object {
      "id": "2",
      "name": "player2",
    },
  ],
  "type": "SET_PLAYERS",
}
`;

exports[`actions should create setRound action 1`] = `
Object {
  "dart": 0,
  "player": 0,
  "round": 0,
  "roundValue": "20-t",
  "type": "SET_ROUND",
}
`;

exports[`actions should create setTotals action 1`] = `
Object {
  "totals": Array [
    301,
  ],
  "type": "SET_TOTALS",
}
`;

exports[`actions should create setWaiting action 1`] = `
Object {
  "type": "SET_WAITING",
  "waiting": true,
}
`;

exports[`actions should create setWinner action 1`] = `
Object {
  "type": "SET_WINNER",
  "winner": 0,
}
`;
